### PR TITLE
coll/tsp: Use pre-completed request for empty schedules

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Save an unnecessary request allocation for trivial operations. Also fixes an incorrect usage of MPID_Request_complete in the MPIR-layer.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpi/coll/transports/gentran/tsp_gentran.c
Comment (zhenggb72) [inline src/mpi/coll/transports/gentran/tsp_gentran.c:589]: Need an error checking for NULL? Otherwise LGTM

**Review by zhenggb72 — APPROVED** [review]

File: src/mpi/coll/transports/gentran/tsp_gentran.c
Comment (raffenet) [reply src/mpi/coll/transports/gentran/tsp_gentran.c:589]: Let me take a look at our other usages. At the moment, MPIR_Request_create_complete always returns a constant handle value, but we may still check for NULL in case that ever changes.

Comment (raffenet) [reply src/mpi/coll/transports/gentran/tsp_gentran.c:589]: Sorry, not a constant handle, but a pointer into a statically allocated builtin request array.

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpi/coll/transports/gentran/tsp_gentran.c
Comment (raffenet) [reply src/mpi/coll/transports/gentran/tsp_gentran.c:589]: Added error check for `NULL`, since it is often used with this API.

**Review by hzhou — APPROVED** [review]
Looks good. Approved as long as tests are clear

**Pull request merged**
